src/hooks/use-pull-push-timestamps.ts

```diff
@@ -0,0 +1,83 @@
+import { __ } from '@wordpress/i18n';
+import { formatDistanceToNow } from 'date-fns';
+import { useCallback } from 'react';
+
+const SYNC_TIMESTAMPS_STORAGE_KEY = 'wp-studio-sync-timestamps';
+
+interface SyncTimestamps {
+	[ localSiteId: string ]: {
+		[ connectedSiteId: number ]: {
+			lastPull?: number;
+			lastPush?: number;
+		};
+	};
+}
+
+export function usePullPushTimestamps() {
+	const getStoredTimestamps = useCallback( (): SyncTimestamps => {
+		try {
+			const stored = localStorage.getItem( SYNC_TIMESTAMPS_STORAGE_KEY );
+			return stored ? JSON.parse( stored ) : {};
+		} catch ( e ) {
+			console.error( 'Failed to parse sync timestamps:', e );
+			return {};
+		}
+	}, [] );
+
+	const updateTimestamp = useCallback(
+		( localSiteId: string, connectedSiteId: number, type: 'pull' | 'push' ) => {
+			try {
+				const timestamps = getStoredTimestamps();
+				timestamps[ localSiteId ] = timestamps[ localSiteId ] || {};
+				timestamps[ localSiteId ][ connectedSiteId ] = {
+					...timestamps[ localSiteId ][ connectedSiteId ],
+					[ type === 'pull' ? 'lastPull' : 'lastPush' ]: Date.now(),
+				};
+				localStorage.setItem( SYNC_TIMESTAMPS_STORAGE_KEY, JSON.stringify( timestamps ) );
+			} catch ( e ) {
+				console.error( 'Failed to update sync timestamp:', e );
+			}
+		},
+		[ getStoredTimestamps ]
+	);
+
+	const getLastSyncTime = useCallback(
+		( localSiteId: string, connectedSiteId: number, type: 'pull' | 'push' ): string => {
+			const timestamps = getStoredTimestamps();
+			const localSiteTimestamps = timestamps[ localSiteId ] || {};
+			const siteTimestamps = localSiteTimestamps[ connectedSiteId ] || {};
+			const timestamp = type === 'pull' ? siteTimestamps.lastPull : siteTimestamps.lastPush;
+
+			if ( ! timestamp ) {
+				return __( 'Never synced' );
```

Comment:
I do not have a strong preference either. By the way, I did updated the wording to `Never pushed` and `Never pulled` for now but it can be no wording as well.

Reply:
I have a similar opinion as Antonio here. If there hasn't been any pull/push before, I would personally remove the tooltip altogether as the `Never pushed` / `Never pulled` information seems a little bit redundant to me.

Reply:
Sounds good, I made changes in https://github.com/Automattic/studio/pull/684/commits/00364756ec5793c2bcbac62f4498d0114973c262

Reply:
Also, would it make sense to change the form of the other message to 2nd person, e.g., 'You pulled this site 1 minute ago'?

Reply:
I will implement this and let's go from there 👍